src/se/element.rs

```diff
@@ -475,9 +482,21 @@ impl<'w, 'k, W: Write> SerializeStruct for Struct<'w, 'k, W> {
             }
         } else {
             self.ser.ser.writer.write_char('>')?;
-            self.ser.ser.writer.write_str(&self.children)?;
+            if self.contains_non_attribute_keys {
+                self.ser.ser.writer.write_str(&self.children)?;
+
+                self.ser.ser.indent.write_indent(&mut self.ser.ser.writer)?;
+            } else {
+                //calculate indentation length
+                let mut indentation = String::new();
+                self.ser.ser.indent.increase();
+                self.ser.ser.indent.write_indent(&mut indentation)?;
+                self.ser.ser.indent.decrease();
 
-            self.ser.ser.indent.write_indent(&mut self.ser.ser.writer)?;
+                let children = self.children.split_at(indentation.len()).1;
+
+                self.ser.ser.writer.write_str(children)?;
+            }
```

Comment:
So the idea is to cut written indentation. I wonder, is it possible to defer intendation write until we will sure that it is necessary?

Reply:
That would require storing the serialized elements structured with their indentation separated instead of just the single string for all child elements. 

I would guess the overhead of storing separated indentation and recombining all these for every struct would be quite large compared to removing indentation in 1 minor case.
I would guess more than 75% of structs use the other if branch that uses the normal behavior.
Since we only care about the case where the TEXT (\<any>_TEXT_\</any>) part is a single statement this was the easiest implementation I could think of. If you have a better idea to get the desired behavior feel free. 

What I would suggest to take a second look is the new format for the tests mentioned above. It is questionable rather the formatting should be this way or rather they require a special case treatment.

Great regards
RedIODev

Reply:
I can't get it to work. The Indentation::current().len() and Indentation::current_indent_len both are off by 1 byte in some but not all cases. I can't see another way to ask Indent for its length.
Great regards
RedIODev

Reply:
Ok, I'll look then myself. Thanks for PR!